doc/acl/SONiC_ACL_based_metering_hld.md

```diff
@@ -0,0 +1,430 @@
+# SONiC ACL based metering Design #
+
+## Table of Content
+
+- [Revision][def]
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High-Level Design](#high-level-design)
+- [SAI API Requirements](#sai-api-requirements)
+- [DB Schema](#db-schema)
+- [SWSS Enhancements](#swss-enhancements)
+- [Configuration and management ](#configuration-and-management)
+    - [Config command](#config-command)
+    - [Show command](#show-command)
+- [YANG Model changes](#yang-model-changes)
+- [Warmboot and Fastboot Considerations](#warmboot-and-fastboot-considerations)
+- [Assumptions/Limitations](#assumptionslimitations)
+- [Testing Design](#testing-design)
+    - [VS Test cases](#vs-test-cases)
+    - [System Test cases](#system-test-cases)
+
+
+### Revision
+
+ | Rev |     Date    |       Author             | Change Description                   |
+ |:---:|:-----------:|:------------------------:|--------------------------------------|
+ | 0.1 |  01/07/24   |  Rajkumar (Marvell)      | Initial version                      |
+
+### Scope
+This document covers the design of ACL based policer. This includes the requirement definition, CLI/yang models, DB schema changes and enhancement to existing acl module in sonic-utils and swss.
+
+### Definitions/Abbreviations
+ ACL - Access Control List
+
+### Overview
+In SONiC, the current design of dataplane ACL supports actions:packet_action, mirror_action. This document proposes the support of new ACL action "policer".
+
+### Requirements
+- Support ACL action "policer" for all DATAPLANE table types. 
+- Allow ACL action policer as configurable option from user. Applicable for rules from all below predefined DATAPLANE ACL tables,
+    - L3
+    - L3V6
+    - L3V4V6
+    - MIRROR
+    - MIRRORV6
+    - MIRRORDSCP
+
+- Support for stage: INGRESS and EGRESS.
```

Comment:
why STAGE_PRE_INGRESS  is not supported?

Reply:
Support for stage PRE/POST INGRESS is not yet added in SONiC acl. I will explicitly capture the same in HLD.